### ISSUE TITLE
Course archive uses heading elements as list items

### ISSUE THREAD

**Issue by mtthwhggns** [apluslms/a-plus#596]
Generally, headings should be used to break down content into shorter blocks.

On the course archive, every list item is an `h3` tag, which isn't very semantic. 

Related: https://apluslms.github.io/accessibility-audit/patterns/fixable-a/semantic-markup-and-sequences/
